File: school-lunch-menu.Windows/Models/Api/FamilyAllergyResponse.cs
using System.Text.Json.Serialization;

namespace SchoolLunchMenu.Models.Api;

/// <summary>
/// An allergen item returned by the LINQ Connect family allergy endpoint.
/// The response is a JSON array of these items.
/// </summary>
public record AllergyItem
{
    /// <summary>
    /// The unique identifier for the allergen.
    /// </summary>
    [JsonPropertyName("AllergyId")]
    public string AllergyId { get; init; } = string.Empty;

    /// <summary>
    /// The sort order used for display purposes.
    /// </summary>
    [JsonPropertyName("SortOrder")]
    public int SortOrder { get; init; }

    /// <summary>
    /// The display name of the allergen.
    /// </summary>
    [JsonPropertyName("Name")]
    public string Name { get; init; } = string.Empty;
}

File: school-lunch-menu.Mac/SchoolLunchMenu/Models/Api/FamilyAllergyResponse.swift
import Foundation

/// An allergen item returned by the LINQ Connect family allergy endpoint.
/// The response is a JSON array of these items.
struct AllergyItem: Codable {
    /// The unique identifier for the allergen.
    let allergyId: String

    /// The sort order used for display purposes.
    let sortOrder: Int

    /// The display name of the allergen.
    let name: String

    enum CodingKeys: String, CodingKey {
        case allergyId = "AllergyId"
        case sortOrder = "SortOrder"
        case name = "Name"
    }

    init(allergyId: String, sortOrder: Int, name: String) {
        self.allergyId = allergyId
        self.sortOrder = sortOrder
        self.name = name
    }
}
